config/setup/tls.go

```diff
@@ -25,6 +53,64 @@ func TLS(c *Controller) (middleware.Middleware, error) {
 			return nil, c.ArgErr()
 		}
 		c.TLS.Key = c.Val()
+
+		// Optional block
+		for c.NextBlock() {
+			switch c.Val() {
+			case "protocols":
+				args := c.RemainingArgs()
+				if len(args) != 2 {
+					return nil, c.ArgErr()
+				}
+				value, ok := supportedProtocols[strings.ToLower(args[0])]
+				if !ok {
+					return nil, c.ArgErr()
+				}
+				c.TLS.ProtocolMinVersion = value
+				value, ok = supportedProtocols[strings.ToLower(args[1])]
+				if !ok {
+					return nil, c.ArgErr()
+				}
+				c.TLS.ProtocolMaxVersion = value
+			case "ciphers":
+				for c.NextArg() {
+					value, ok := supportedCiphers[strings.ToUpper(c.Val())]
+					if !ok {
+						return nil, c.ArgErr()
+					}
+					c.TLS.Ciphers = append(c.TLS.Ciphers, value)
+				}
+			case "cache":
+				if !c.NextArg() {
+					return nil, c.ArgErr()
+				}
+				c.TLS.CacheSize, _ = strconv.Atoi(c.Val())
```

Comment:
Instead of ignoring the error, let's return an error so the user can know that their argument must be a number.